File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.cpp
//{ Driver Code Starts
// Initial Template for C++
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function Template for C++
class Solution {
  public:
    // Function to find the sum of contiguous subarray with maximum sum.
    int maxSubarraySum(vector<int> &arr) {
        // code here...
        int maxi = INT_MIN;
        int sum = 0;
        int ans= -1;
        int start =0;
        int ansEnd = -1;
        for(int i = 0;i<arr.size();i++){
            if(sum==0){
                start = i;
            }
            sum = sum + arr[i];
            if(sum>maxi){
                maxi = sum;
                ans = start;
                ansEnd = i;
            }
            if(sum<0){
                sum = 0;
            }
        }
        return maxi;
    }
};


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    cin.ignore(); // To discard any leftover newline characters
    while (t--)   // while testcases exist
    {
        vector<int> arr;
        string input;
        getline(cin, input); // Read the entire line for the array elements
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }

        Solution ob;
        cout << ob.maxSubarraySum(arr) << endl;
    }
}
// } Driver Code Ends

File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.py
#User function Template for python3

class Solution:
    ##Complete this function
    #Function to find the sum of contiguous subarray with maximum sum.
    def maxSubArraySum(self,arr):
        ##Your code here    
        max_ending_here = arr[0]  
        max_so_far = arr[0]     

        for i in range(1, len(arr)):
            
            max_ending_here = max(arr[i], max_ending_here + arr[i])
           
            max_so_far = max(max_so_far, max_ending_here)

        return max_so_far
#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math


def main():
    T = int(input())
    while (T > 0):

        arr = [int(x) for x in input().strip().split()]

        ob = Solution()

        print(ob.maxSubArraySum(arr))

        T -= 1


if __name__ == "__main__":
    main()

# } Driver Code Ends